### PR TITLE
Feat: SetHeaders additional logging info

### PR DESCRIPTION
**Pull request by NickKhalow**
## What does this PR change?

https://github.com/decentraland/unity-explorer/issues/2761
Due there are no repro steps and there is very little info to debug the issue we should add the additional context

Added: 
1. More context for InvalidOperationException
2. Safety net for WebRequestHeader's ctor

## How to test the changes?

1. Launch the explorer
3. Play happy path

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by AnsisMalins — APPROVED** [review]
I'm approving this change because it does not make things worse, but please consider my comments.

File: Explorer/Assets/DCL/WebRequests/WebRequestHeader.cs [hunk 1]
```diff
@@ -7,11 +9,23 @@ public readonly struct WebRequestHeader
...
+            Name = ValueOrEmpty(name);
+            Value = ValueOrEmpty(value);
         }
 
         public override string ToString() =>
             $"WebRequestHeader({Name}: {Value})";
+
+        private static string ValueOrEmpty(string value)
+        {
+            // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
```
[the remark this answers is not in the document]
Comment (NickKhalow) [reply Explorer/Assets/DCL/WebRequests/WebRequestHeader.cs:21]: It is not a case of this PR. There are many precedents of "// ReSharper disable" in the codebase. If you want to change it please open a ticket with the suggestion.

File: Explorer/Assets/DCL/WebRequests/WebRequestHeader.cs [hunk 2]
```diff
@@ -7,11 +9,23 @@ public readonly struct WebRequestHeader
...
+            if (value == null)
+            {
+                ReportHub.LogError(ReportCategory.ECS, "Propagated null value in WebRequestHeader.");
```
[the remark this answers is not in the document]
Comment (NickKhalow) [reply Explorer/Assets/DCL/WebRequests/WebRequestHeader.cs:24]: According to the ctor's contract it's not supposed to accept a null value, therefore `null` is considered as an error in this case

File: Explorer/Assets/DCL/WebRequests/RequestEnvelope.cs [hunk 3]
```diff
@@ -111,7 +111,9 @@ private void SetHeaders(UnityWebRequest unityWebRequest)
             for (var i = 0; i < count; i++)
             {
                 WebRequestHeader header = info[i];
-                unityWebRequest.SetRequestHeader(header.Name, header.Value);
+
+                try { unityWebRequest.SetRequestHeader(header.Name, header.Value); }
+                catch (InvalidOperationException e) { throw new Exception($"Cannot set header: {header.Name} - {header.Value}", e); }
```
[the remark this answers is not in the document]
Comment (NickKhalow) [reply Explorer/Assets/DCL/WebRequests/RequestEnvelope.cs:116]: `The innerExteption never gets printed out by any logging system` is wrong. Rethrown errors are printed to the StackTrace with "Rethrow as" key.

**Review by DafGreco — APPROVED** [review]
Sanity check performed in both platforms ✅ 

- Teleport to different scenes with music and auto-emoting
- Metadynelabs 
- Backpack
- Emotes 
- Chat interactions

**Pull request merged**